frontend/assets/style/_typography.scss

```diff
@@ -75,43 +75,52 @@ $lineHeight: 3, 2.25, 1.6875, 1.3125, 1;
   }
 }
 
+// Responsive Headings:
+// NOTE: When we mix a h1/h2 with .is-[n], (ex: "h1.is-3") it will trigger unexpected behaviors:
+// On mobile/tablet the element will have different font-size (see rules bellow)
+
+// Well, I don't know any clean way of fixing it with CSS.. but here's a trick:
+// .is-3(role="heading" aria-level="1")
+// This has exactly the same semantic meaning has h1 tag.
+
+// Other solution could be to completely remove styles directly applied to h# tag
+// and always add the CSS class to the heading. (ex: h1.is-1).
 @include until($tablet) {
-  .is-1,
-  h1 {
+  h1,
+  // h1:not([class^=".is-"]), // Option 3: - This also fixes the bug!
+  // It other words, it means: "apply this to a h1 that doesn't contain .is-[n] class.
+  .is-1 {
```

Comment:
cc @pieer and @taoeffect

My option here would be to completely remove the styles added directly to h# tags and always add .is-[n] to a heading. I prefer to avoid coupling styling with semantics to avoid two things: 1. this type of bugs and 2. misusage of heading tags.
The trick with role="heading" is funny, but overkilling. Each time we would need to do it, probably we would forget it until someone notices the bug.
Using h1:not([class^=".is-"]) fixes it, but it's a reflection of our lazy/simple approach that doesn't style properly a heading with a .is-[n]. Besides that, it increased its CSS specificity and it would make our life harder if we need to override it.

Reply:
I agree with you Sandrina